// gatsby-source-sanity/lib/util/errorMap.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ERROR_MAP = exports.CODES = exports.prefixId = exports.pluginPrefix = void 0;
exports.pluginPrefix = 'gatsby-source-sanity';
function prefixId(id) {
    return `${exports.pluginPrefix}_${id}`;
}
exports.prefixId = prefixId;
var ReporterLevel;
(function (ReporterLevel) {
    ReporterLevel["Error"] = "ERROR";
})(ReporterLevel || (ReporterLevel = {}));
var ReporterCategory;
(function (ReporterCategory) {
    // Error caused by user (typically, site misconfiguration)
    ReporterCategory["User"] = "USER";
    // Error caused by Sanity plugin ("third party" relative to Gatsby Cloud)
    ReporterCategory["ThirdParty"] = "THIRD_PARTY";
    // Error caused by Gatsby process
    ReporterCategory["System"] = "SYSTEM";
})(ReporterCategory || (ReporterCategory = {}));
exports.CODES = {
    UnsupportedGatsbyVersion: '10000',
    SchemaFetchError: '10001',
    MissingProjectId: '10002',
    MissingDataset: '10002',
};
exports.ERROR_MAP = {
    [exports.CODES.UnsupportedGatsbyVersion]: {
        text: (context) => context.sourceMessage,
        level: ReporterLevel.Error,
        category: ReporterCategory.User,
    },
    [exports.CODES.SchemaFetchError]: {
        text: (context) => context.sourceMessage,
        level: ReporterLevel.Error,
        category: ReporterCategory.ThirdParty,
    },
    [exports.CODES.MissingProjectId]: {
        text: (context) => context.sourceMessage,
        level: ReporterLevel.Error,
        category: ReporterCategory.User,
    },
    [exports.CODES.MissingDataset]: {
        text: (context) => context.sourceMessage,
        level: ReporterLevel.Error,
        category: ReporterCategory.User,
    },
};
